### PR TITLE
Improve clarity of Terraform Enterprise VCS docs

### PR DESCRIPTION
**Pull request by jlsuttles**
There has been some confusion from customers trying to get the GitLab and Bitbucket Cloud integrations setup. These changes are meant to make the process more clear.

### CODE REVIEW

File: source/docs/enterprise/vcs/bitbucket.html.md [hunk 1]
```diff
@@ -13,46 +13,49 @@ Additionally, plans are run when a pull request is created or updated. Terraform
...
 providing quick feedback on proposed changes.
 
-## Registering an OAuth Application & Client
+## Setup an Organization to use Bitbucket Cloud
 
-### Creating and Updating a Bitbucket Cloud OAuth Application
+### Create a Bitbucket Cloud OAuth Consumer
 
-You will need to register Terraform Enterprise as an OAuth Application within your Bitbucket Cloud account. Proceed to https://bitbucket.org/account/user/your-username/oauth-consumers/new. Fill out the required information and set the Redirect URI to a placeholder (ie: http://example.com), as you will need to register the Bitbucket Client with Terraform Enterprise prior to receiving this value. Check all of the permission fields that apply to you, and click Save
+You need to register Terraform Enterprise as an OAuth consumer within your Bitbucket Cloud account. Proceed to https://bitbucket.org/account/user/your-username/oauth-consumers/new. Fill out the form with the following information.
```
Comment (feministy) [inline source/docs/enterprise/vcs/bitbucket.html.md]: can we make this into a link for folks to click on? even if the full URL is the text, having a clickable link is 💯

File: source/docs/enterprise/vcs/bitbucket.html.md [hunk 2]
```diff
@@ -13,46 +13,49 @@ Additionally, plans are run when a pull request is created or updated. Terraform
...
+The following **Permissions** are required:
 
-In a new tab, navigate to https://atlas.hashicorp.com/settings and, in the left-side panel, select the organization that you’d like to administer your Bitbucket connection, then click on “configuration” in the left-side panel.
+- **Account**: Email, Read, Write
+- **Repositories**: Read, Write, Admin
+- **Pull requests**: Read, Write
+- **Webhooks**: Read and write
 
-Within the “Add OAuthClient” pane, select Bitbucket Cloud and fill in your application key and secret. In the base url field, enter the root url of your Bitbucket instance (i.e. https://bitbucket.org). In the API url field, enter the base api url (i.e. https://api.bitbucket.org/2.0). Create the OAuth client.
+Upon saving, you will be redirected to https://bitbucket.org/account/user/your-username/api. Scroll down to the **OAuth consumers** section and click on the consumer you just created. Copy the **Key** and **Secret**. You will need to enter these values into Terraform Enterprise. Leave this tab open in your browser as you will need to return to it in a moment.
```
Comment (feministy) [inline source/docs/enterprise/vcs/bitbucket.html.md]: any time it is more than 2 steps, i prefer to use a numbered list so folks don't accidentally miss one. maybe formatted like this?

```
Upon saving, you will be redirected to https://bitbucket.org/account/user/your-username/api. 

1. Scroll down to the **OAuth consumers** section and click on the consumer you just created
2. Copy the **Key** and **Secret**
3. Enter these values into Terraform Enterprise <<NAME OF PAGE>> 

Leave this tab open in your browser as you will need to return to it in a moment.
```

File: source/docs/enterprise/vcs/gitlab.html.md [hunk 3]
```diff
@@ -14,42 +14,44 @@ Additionally, plans are run when a pull request is created or updated. Terraform
...
+Scroll down to the **Add OAuthClient** pane and fill out the form with the following information.
 
-Once you have created your client, you will be redirected back to the configurations page for your chosen organization. On that page, find the “OAuth Clients” pane and copy the `Callback URL` for your GitLab OAuth Client. In a new tab, navigate back to https://gitlab.com/profile/applications select the terraform-enterprise OAuth Application and click edit. Enter the `Callback URL` you just copied in the field labeled `Redirect URI`. Save the application.
+- **Oauth client**: Your GitLab installation type (e.g. GitLab.com, GitLab Community Edition, or GitLab Enterprise)
+- **Application key**: Application Id (from GitLab in the previous section)
+- **Base url**: https://gitlab.com (for GitLab.com)
```
Comment (feministy) [inline source/docs/enterprise/vcs/gitlab.html.md]:
```
- **Base url**: https://gitlab.com (for GitLab.com; If you are using GitLab Community Edition or GitLab Enterprise your URL will be different)
```

File: source/docs/enterprise/vcs/gitlab.html.md [hunk 4]
```diff
@@ -14,42 +14,44 @@ Additionally, plans are run when a pull request is created or updated. Terraform
...
+- **Api url**: https://gitlab.com/api/v3 for GitLab.com
```
Comment (feministy) [inline source/docs/enterprise/vcs/gitlab.html.md]: would add same caveat here about URLs as well

File: source/docs/enterprise/vcs/bitbucket.html.md (hunk 1 above)
Comment (jlsuttles) [inline source/docs/enterprise/vcs/bitbucket.html.md]: This happens during compilation, URLS are automagically link-ified! 🔗

**Comment by jlsuttles** [comment]
@apparentlymart Thanks! Ready to :shipit:

**Comment by apparentlymart** [comment]
I cherry-picked this into the old (<= 0.10) website branch in the Terraform repo and deployed it, and I'm seeing the changes:

https://www.terraform.io/docs/enterprise/vcs/bitbucket.html
https://www.terraform.io/docs/enterprise/vcs/gitlab.html

(Sometimes it takes a hard-refresh to see updates to the site if you've been there before because we have some strong caching headers there.)

Please let me know if anything seems amiss!

File: source/docs/enterprise/vcs/bitbucket.html.md (hunk 1 above)
Comment (feministy) [inline source/docs/enterprise/vcs/bitbucket.html.md]: this is amazing

**Pull request merged**